### PR TITLE
Duplo 11508 RDS secret manager attribute bug fix

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Bug fix attribute store_details_in_secret_manager not getting set at duplo
## Summary of changes
Setting values for attribute store_details_in_secret_manager on resource create context and fetching value on resource read context for duplocloud_rds_instance
This PR does the following:

- Setting DuploRdsInstance.StoreDetailsInSecretManager in rdsInstanceFromState()
- Setting store_details_in_secret_manager in rdsInstanceToState()
## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ✓] Manually, on a remote test system

## Describe any breaking changes

- ...

___

## **Type**
bug_fix, enhancement

___

## **Description**
- Introduced a new Terraform resource `duplocloud_gcp_k8_node_pool` for managing GCP Kubernetes Node Pools with full CRUD operations.
- Fixed an issue where `store_details_in_secret_manager` attribute was not being set correctly for `duplocloud_rds_instance` resource.
- Added SDK support for GCP Kubernetes Node Pool operations, including creating, retrieving, updating, and deleting node pools.
- Updated CHANGELOG to reflect the addition of the GCP Node Pool resource and the fix for the RDS instance attribute.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>provider.go</strong><dd><code>Register GCP Node Pools Resource in Provider</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/provider.go

- Added `duplocloud_gcp_node_pools` to the provider's resources.

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/419/files#diff-0959744faba70cfc67abd70733e39a6f668d41c9fae69a27968841ef8ed56eeb">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_gcp_k8_node_pool.go</strong><dd><code>Implement GCP Kubernetes Node Pool Resource</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/resource_duplo_gcp_k8_node_pool.go

<li>Introduced a new resource <code>duplocloud_gcp_k8_node_pool</code> with full CRUD <br>operations.<br> <li> Implemented schema definition and CRUD functions for managing GCP <br>Kubernetes Node Pools.<br> <li> Added support for various configurations like auto-scaling, upgrade <br>settings, and node taints.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/419/files#diff-bc0cc4a27038eb399598c526fd7148bd7cd9349310b06235a990a5f2cac1eaf9">+735/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gcp_node_pools.go</strong><dd><code>Add SDK Support for GCP Kubernetes Node Pools</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplosdk/gcp_node_pools.go

<li>Added SDK support for GCP Kubernetes Node Pool operations.<br> <li> Implemented functions for creating, retrieving, updating, and deleting <br>GCP Kubernetes Node Pools.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/419/files#diff-d7d6988aa1383fd5e2f3442050fe77b783a7d1a7094e70e20982f64d9f1c5b03">+129/-0</a>&nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Bug_fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_rds_instance.go</strong><dd><code>Fix store_details_in_secret_manager Attribute Handling</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/resource_duplo_rds_instance.go

<li>Fixed the <code>store_details_in_secret_manager</code> attribute not being set <br>correctly.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/419/files#diff-a8bf9b56a3e9b39899a5d37d7608110678908754e0848fab2d0320653ea65f71">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update CHANGELOG for GCP Node Pool Resource and RDS Instance Fix</code></dd></summary>
<hr>
      
CHANGELOG.md

<li>Updated CHANGELOG with entries for the new <code>duplocloud_gcp_node_pool</code> <br>resource and the fix for <code>store_details_in_secret_manager</code>.<br>

</details>
    

  </td>
  <td><a href="https:/duplocloud/terraform-provider-duplocloud/pull/419/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-11508 TF: RDS Save Password in SecretsManager Doesn't actually work](https://app.clickup.com/t/8655600/DUPLO-11508)

File: duplocloud/resource_duplo_gcp_k8_node_pool.go [hunk 1]
```diff
@@ -0,0 +1,735 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func gcpK8NodePoolFunctionSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"tenant_id": {
+			Description:  "The GUID of the tenant that the cloud function will be created in.",
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_gcp_k8_node_pool.go:20]: Please change the description. Add node pool instead of cloud function.

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_gcp_k8_node_pool.go:20]: done

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_gcp_k8_node_pool.go:20]: I will change here too

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_gcp_k8_node_pool.go]: done

**Review by joek-duplo — CHANGES_REQUESTED** [review]
This PR should not have any code for node pools at all. Please remove that and fix the changelog.

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**